handlers/member.go

```diff
@@ -0,0 +1,83 @@
+package handlers
+
+import (
+	"log"
+	"net/http"
+	"net/url"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/coreos/etcd/client"
+	"github.com/gorilla/mux"
+	ctx "golang.org/x/net/context"
+)
+
+const (
+	// DefaultClientTimeout is the default timeout for etcd client
+	DefaultClientTimeout = 5 * time.Second
+)
+
+// MemberHandler can be used to query the latest etcd quorum members in a cleaner format compare to the standard etcd
+// format provided by GET "/{token:[a-f0-9]{32}}"
+func MemberHandler(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	token := vars["token"]
+	log.Println("token: ", token)
+
+	cfg := client.Config{
+		Endpoints:               etcdMachines,
+		Transport:               client.DefaultTransport,
+		HeaderTimeoutPerRequest: DefaultClientTimeout,
+	}
+
+	c, err := client.New(cfg)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+	}
+
+	key := path.Join("_etcd", "registry", token)
+	kAPI := client.NewKeysAPI(c)
+
+	resp, err := kAPI.Get(ctx.Background(), key, &client.GetOptions{Recursive: true, Quorum: true})
```

Comment:
Ok, if the discovery service relies on the etcd store, how do we keep it stable if a member goes down? we don't have a castle cluster in global that can automate the failover for us.